src/test/java/com/vinyl/DeleteProductFromCartTest.java

```diff
@@ -0,0 +1,128 @@
+package com.vinyl;
+
+import com.vinyl.helper.TokenHeaderHelper;
+import com.vinyl.model.Cart;
+import com.vinyl.model.Product;
+import com.vinyl.model.ProductCart;
+import com.vinyl.model.User;
+import com.vinyl.repository.CartRepository;
+import com.vinyl.repository.ProductCartRepository;
+import com.vinyl.repository.TokenRepository;
+import org.assertj.core.api.Assertions;
+import org.assertj.core.util.Lists;
+import org.junit.Test;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.*;
+
+import java.time.LocalDate;
+
+public class DeleteProductFromCartTest extends BaseIntegration {
+
+    @Autowired
+    private CartRepository cartRepository;
+
+    @Autowired
+    private TokenHeaderHelper tokenHeaderHelper;
+
+    @Autowired
+    private ProductCartRepository productCartRepository;
+
+    @Autowired
+    private TokenRepository tokenRepository;
+
+    private Product product;
+
+    @Override
+    public void setUp() {
+        super.setUp();
+        product = defaultEntitiesHelper.createProduct();
+    }
+
+    @Test
+    public void testWhenUserIsLoggedInAndCartExists() {
+        Cart cart = cartSetup();
+
+        ResponseEntity<?> response = setUpHeaderAndGetTheResponse();
+
+        Assertions.assertThat(response.getStatusCode()).isEqualTo(HttpStatus.OK);
+        Assertions.assertThat(productCartRepository.findByProductAndCart(product, cart)).isNull();
+    }
+
+    @Test
+    public void testWhenUserIsLoggedInAndCartDoesNotExist() {
+        ResponseEntity<?> response = setUpHeaderAndGetTheResponse();
+        Assertions.assertThat(response.getStatusCode()).isEqualTo(HttpStatus.BAD_REQUEST);
+    }
+
+    @Test
+    public void testWhenUserIsLoggedInAndTriesToDeleteFromAnotherCart() {
+        cartSetup();
+
+        User otherUser = createUser("otheruser@email.com");
+        HttpHeaders headers = tokenHeaderHelper.setupToken(token.getHash());
+        ResponseEntity<?> response = trt.exchange("/users/" + otherUser.getId() + "/" + product.getId(), HttpMethod.POST, new HttpEntity<>(headers), Void.class);
+
+        Assertions.assertThat(response.getStatusCode()).isEqualTo(HttpStatus.UNAUTHORIZED);
+    }
+
+    @Test
+    public void testWhenProductToDeleteIsInvalid() {
+        defaultEntitiesHelper.createCart(user);
+        HttpHeaders headers = tokenHeaderHelper.setupToken(token.getHash());
+
+        Product otherProduct = createOtherProduct();
+        ResponseEntity<?> response = trt.exchange("/users/" + user.getId() + "/" + otherProduct.getId(), HttpMethod.POST, new HttpEntity<>(headers), Void.class);
+
+        Assertions.assertThat(response.getStatusCode()).isEqualTo(HttpStatus.BAD_REQUEST);
+    }
+
+    @Test
+    public void testWhenTokenIsMissing() {
```

Comment:
these tests with token expired, token missing, etc will be valid for all endpoints that pertain to a logged in used. Create another abstract test class maybe called LoggedInBaseIntegration that contains these tests, the new class will extend the base class and it will be extended by all other classes that need the token tests

Reply:
done